src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerService.java

```diff
@@ -113,6 +117,12 @@ public void updateGeneralPartner(String generalPartnerId, GeneralPartnerDataDto
         repository.save(generalPartnerDaoAfterPatch);
     }
 
+    public GeneralPartnerDto getGeneralPartner(Transaction transaction, String submissionId) throws ServiceException {
+        checkGeneralPartnerIsLinkedToPartnership(transaction, submissionId);
```

Comment:
the call to checkGeneralPartnerIsLinkedToPartnership should pass generalPartnerId instead of submissionID

Reply:
As above, updated parameter name.